### DISCUSSION TITLE
Cannot generate one-hot format labels because not set `num_classes` in `data_preprocessor`

### DISCUSSION THREAD

**Discussion by zeynepozdemir** [open-mmlab/mmclassification#1284] [Q&A]
Hello, I am trying to use the eva model with a custom dataset of 20 classes. I could not find the reason for the error in the image below.
![image](https://user-images.githubusercontent.com/6259137/209923948-b8d29353-e3e1-4a01-b128-713eb79725a9.png)

**Comment by yingfhu** [comment]
Please find the `data_preprocessor` field in your config file and add `num_classes=20` in it, e.g.
```
data_preprocessor = dict(
    num_classes=20,
    ...
```

**Comment by zpz915** [comment]
我也是  我加了分类头  也不管用
